examples/nft-nd-ns/ns.go

```diff
@@ -0,0 +1,655 @@
+	return resolve(ctx, name, typ, 2)
+}
+
+// updateBalance updates account's balance and account's tokens.
+func updateBalance(ctx storage.Context, tokenId []byte, acc interop.Hash160, diff int) {
+	balanceKey := append([]byte{prefixBalance}, acc...)
+	var balance int
+	if b := storage.Get(ctx, balanceKey); b != nil {
+		balance = b.(int)
+	}
+	balance += diff
+	if balance == 0 {
+		storage.Delete(ctx, balanceKey)
+	} else {
+		storage.Put(ctx, balanceKey, balance)
+	}
+
+	tokenKey := getTokenKey(tokenId)
+	accountTokenKey := append(append([]byte{prefixAccountToken}, acc...), tokenKey...)
+	if diff < 0 {
+		storage.Delete(ctx, accountTokenKey)
+	} else {
+		storage.Put(ctx, accountTokenKey, tokenId)
+	}
+}
+
+// postTransfer sends Transfer notification to the network and calls onNEP11Payment
+// method.
+func postTransfer(from, to interop.Hash160, tokenID []byte, data interface{}) {
+	runtime.Notify("Transfer", from, to, 1, tokenID)
+	if management.GetContract(to) != nil {
+		contract.Call(to, "onNEP11Payment", contract.All, from, 1, tokenID, data)
+	}
+}
+
+// getTotalSupply returns total supply from storage.
+func getTotalSupply(ctx storage.Context) int {
+	val := storage.Get(ctx, []byte{prefixTotalSupply})
+	if val != nil {
+		return val.(int)
+	}
+	return 0
+}
+
+// updateTotalSupply adds specified diff to the total supply.
+func updateTotalSupply(ctx storage.Context, diff int) {
+	tsKey := []byte{prefixTotalSupply}
+	ts := getTotalSupply(ctx)
+	storage.Put(ctx, tsKey, ts+diff)
+}
+
+// getTokenKey computes hash160 from the given tokenID.
+func getTokenKey(tokenID []byte) []byte {
+	return crypto.Ripemd160(tokenID)
+}
+
+// getNameState returns domain name state by the specified tokenID.
+func getNameState(ctx storage.Context, tokenID []byte) NameState {
+	tokenKey := getTokenKey(tokenID)
+	return getNameStateWithKey(ctx, tokenKey)
+}
+
+// getNameStateWithKey returns domain name state by the specified token key.
+func getNameStateWithKey(ctx storage.Context, tokenKey []byte) NameState {
+	nameKey := append([]byte{prefixName}, tokenKey...)
+	nsBytes := storage.Get(ctx, nameKey)
+	if nsBytes == nil {
+		panic("token not found")
+	}
+	ns := std.Deserialize(nsBytes.([]byte)).(NameState)
+	ns.ensureNotExpired()
+	return ns
+}
+
+// putNameState stores domain name state.
+func putNameState(ctx storage.Context, ns NameState) {
+	tokenKey := getTokenKey([]byte(ns.Name))
+	putNameStateWithKey(ctx, tokenKey, ns)
+}
+
+// putNameStateWithKey stores domain name state with the specified token key.
+func putNameStateWithKey(ctx storage.Context, tokenKey []byte, ns NameState) {
+	nameKey := append([]byte{prefixName}, tokenKey...)
+	nsBytes := std.Serialize(ns)
+	storage.Put(ctx, nameKey, nsBytes)
+}
+
+// getRecord returns domain record.
+func getRecord(ctx storage.Context, tokenId []byte, name string, typ RecordType) string {
+	recordKey := getRecordKey(tokenId, name, typ)
+	record := storage.Get(ctx, recordKey)
+	if record == nil {
```

Comment:
If we want to return nil, we can also return `return storage.Get(..).(string)`. I believe the compiler doesn't emit conversions for syscall+type assertion, so we just return the result of `storage.Get` as in C# node.